### PR TITLE
Use eks:cluster-name as clusterId

### PR DESCRIPTION
**Pull request by groodt**
*Issue #775*

*Description of changes:*
Prioritise `eks:cluster-name` as clusterId for cni-metrics-helper. This is the tag that EKS node-groups is automatically adding.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/publisher/publisher.go [hunk 1]
```diff
@@ -207,6 +198,24 @@ func (p *cloudWatchPublisher) getCloudWatchMetricNamespace() *string {
...
+			clusterID, err = ec2Client.GetClusterTag("Name")
+			if err != nil || clusterID == "" {
+				glog.Errorf("Failed to determine cluster-id, defaulting to 'k8s-cluster'.  %v", err)
+				clusterID = "k8s-cluster"
+			}
+		}
+	}
+	glog.Info("Using cluster ID ", clusterID)
+	return clusterID
+}
```
Comment (jaypipes) [inline pkg/publisher/publisher.go]: Hi @groodt thanks very much for the contribution! May I suggest reworking the above to be a little cleaner? You can use a const to list the tag names to try, in order, and then loop through the tag names like so:

```go
const (
        defaultClusterID = "k8s-cluster"
        clusterIDTags = []string{
                "eks:cluster-name",
                "CLUSTER_ID",
                "Name",
        }
)
...
func getClusterID(ec2Client *ec2wrapper.EC2Wrapper) string {
        var clusterID string
        var err error
        for _, tag := range clusterIDTags {
	        clusterID, err = ec2Client.GetClusterTag(tag)
	        if err != nil && clusterID != "" {
                        break
		}
	}
        if clusterID == "" {
                clusterID = defaultClusterID
        }
	glog.Info("Using cluster ID ", clusterID)
	return clusterID
}
```

Comment (groodt) [reply pkg/publisher/publisher.go]: Thanks for the suggestion. That is cleaner. I'll make the changes ASAP.

Comment (groodt) [reply pkg/publisher/publisher.go]: I've made the change. Slices can't be `const`, so I've inlined the variables into the function. Open to alternatives. Just let me know what would be the preference.

Comment (jaypipes) [reply pkg/publisher/publisher.go]: Yes, forgot about that. You could have either just replaced `const` with `var`, but this is perfectly fine as well :)

File: pkg/publisher/publisher.go [hunk 2]
```diff
@@ -207,6 +198,32 @@ func (p *cloudWatchPublisher) getCloudWatchMetricNamespace() *string {
 	return aws.String(cloudwatchMetricNamespace)
 }
 
+func getClusterId(ec2Client *ec2wrapper.EC2Wrapper) string {
```
Comment (jaypipes) [inline pkg/publisher/publisher.go]: Tiny nit... could you name this `getClusterID()` (capitalized ID) please?

**Comment by groodt** [comment]
Changes in. PTAL.

**Comment by groodt** [comment]
@jaypipes 

I just realised that I think the condition should be:
```
		if err == nil && clusterID != "" {
			break
		}
```

**Comment by groodt** [comment]
Thanks @jaypipes 

Will this get merged by a collaborator automatically at some stage, or do I need to assign somebody specific?

**Pull request merged**